core/storage/classifier/gae_models.py

```diff
@@ -181,6 +181,25 @@ def create(
         training_job_instance.put()
         return instance_id
 
+    @classmethod
+    def query_training_jobs(cls, inlcude_deleted=False):
```

Comment:
Updated.
In the case of all failed jobs, Wouldn't it create infinite loop? Assuming the cursor will return None at the end of all jobs?